pkg/network/server.go

```diff
@@ -902,6 +906,13 @@ func (s *Server) requestBlocksOrHeaders(p Peer) error {
 		return fmt.Errorf("%w: %w", errBlocksRequestFailed, err)
 	}
 	if requestMPTNodes {
+		if s.config.NeoFSStateSyncExtensions {
+			nodes, err := s.syncHeaderFetcher.GetMPTNodes()
+			if err != nil {
+				return err
+			}
+			return s.stateSync.AddMPTNodes(nodes)
```

Comment:
I see the same pattern as it was used previously in `upload-state`: you're trying to retrieve all MPT nodes at once. What we need instead is to open a stream, download MPT nodes and add them to statesync module one-by-one. You need to refactor `stateSync.AddMPTNodes` for this, it should accept a function instead of a raw set of nodes.